### PR TITLE
c7n_tencentcloud - resources - cdb & cdb_backup

### PR DESCRIPTION
**Pull request by 723514372466**

### CODE REVIEW

File: tools/c7n_tencentcloud/c7n_tencentcloud/resources/cdb.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# SPDX-License-Identifier: Apache-2.0
+from c7n_tencentcloud.provider import resources
+from c7n_tencentcloud.query import ResourceTypeInfo, QueryResourceManager
+from c7n_tencentcloud.utils import PageMethod, isoformat_datetime_str
+import pytz
+
+
+@resources.register("cdb")
+class CDB(QueryResourceManager):
+    """cdb"""
```
Comment (kapilt) [inline tools/c7n_tencentcloud/c7n_tencentcloud/resources/cdb.py:11]: please put the full name of the service here, the acronyms aren't clearly obvious to, even then I wonder if we just want to have the type named db-instance here for example to make it more obvious.

Comment (723514372466) [reply tools/c7n_tencentcloud/c7n_tencentcloud/resources/cdb.py]: optimized

File: tools/c7n_tencentcloud/c7n_tencentcloud/resources/mysql.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        taggable = True
+
+        datetime_fields_format = {
+            "CreateTime": ("%Y-%m-%d %H:%M:%S", pytz.timezone("Asia/Shanghai"))
+        }
+
+    def augment(self, resources):
+        for resource in resources:
+            cli = self.get_client()
+            resp = cli.execute_query("DescribeDBInstanceInfo",
```
Comment (kapilt) [inline tools/c7n_tencentcloud/c7n_tencentcloud/resources/mysql.py:35]: can we move this to an encryption filter so we don't need to fetch it unless its being used for a policy assertion.

also generally its preferable when adding additional attributes to use a c7n: prefix annotation to distinguish from native ones. we generally make exceptions for tags, but beyond that its generally a legacy motion.

Comment (ritch2022) [reply tools/c7n_tencentcloud/c7n_tencentcloud/resources/mysql.py:35]: Hi kapilt, use c7n: prefix is ok.
Do you have any definition for native attributes and additional attributes?
In our mindset, those directly from API are native, those generated by code are additional.

Comment (kapilt) [reply tools/c7n_tencentcloud/c7n_tencentcloud/resources/mysql.py:35]: Yes, that's accurate, thanks

Comment (ritch2022) [reply tools/c7n_tencentcloud/c7n_tencentcloud/resources/mysql.py:35]: According to this rule, I think there are many codes need to be modified.

Comment (723514372466) [reply tools/c7n_tencentcloud/c7n_tencentcloud/resources/mysql.py]: Added encryption filter

File: tools/c7n_tencentcloud/c7n_tencentcloud/resources/mysql_backup.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+class MySQLBackUp(QueryResourceManager):
+    """mysql-backup"""
+
+    class resource_type(ResourceTypeInfo):
+        """resource_type"""
+        id = "InstanceId"
+        endpoint = "cdb.tencentcloudapi.com"
+        service = "cdb"
+        version = "2017-03-20"
+        enum_spec = ("DescribeDBInstances", "Response.Items[]", {})
```
Comment (thisisshi) [inline tools/c7n_tencentcloud/c7n_tencentcloud/resources/mysql_backup.py:19]: this seems problematic but given that there doesnt seem to be any other way to get backups not associated with an instance (e.g. instance was deleted). I see that the console itself is querying for backups this way:

```
curl 'https://intl-db-api.tencentcloud.com/api/server-cloud3/supeross/service.backup.DescribeBackups' \
-X 'POST' \
-H 'Content-Type: application/json' \
-H 'Accept: application/json' \
-H 'Accept-Language: en-US,en;q=0.9' \
-H 'Accept-Encoding: gzip, deflate, br' \
-H 'Origin: https://console.tencentcloud.com' \
-H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.3 Safari/605.1.15' \
-H 'Referer: https://console.tencentcloud.com/mysql/backup/list/data' \
-H 'Content-Length: 168' \
-H 'Connection: keep-alive' \
-H 'Host: intl-db-api.tencentcloud.com' \
-H 'X-SeqId: 093e6a80-594e-11ed-92d8-ef39972cfd7e' \
-H 'X-CsrfCode: 1366196663' \
-H 'X-Requested-With: XMLHttpRequest' \
-H 'X-QCloud-User-ID: 1307793390' \
-H 'X-QCloud-Transaction-ID: 093e6a80-594e-11ed-92d8-ef39972cfd7e' \
--data-binary '{"Region":"na-ashburn","Limit":20,"Offset":0,"ApiModule":"backupService","Product":"mysql","OrderBy":"Date","OrderDirection":"DESC","MaxStartTime":"","MinStartTime":""}'
```

**Review by thisisshi — CHANGES_REQUESTED** [review]

File: tools/c7n_tencentcloud/c7n_tencentcloud/resources/mysql.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+            field_format = self.resource_type.datetime_fields_format["CreateTime"]
+            resource["CreateTime"] = isoformat_datetime_str(resource["CreateTime"],
+                                                            field_format[0],
+                                                            field_format[1])
+        return resources
+
+
+@MySQL.filter_registry.register('encryption')
+class EncryptionFilter(Filter):
+    schema = type_schema('encryption', value={'type': 'string'})
```
[the remark this answers is not in the document]
Comment (723514372466) [reply tools/c7n_tencentcloud/c7n_tencentcloud/resources/mysql.py]: optimized

File: tools/c7n_tencentcloud/c7n_tencentcloud/resources/mysql.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+    def process(self, resources, event=None):
+        value = self.data.get('value', "YES")
```
[the remark this answers is not in the document]
Comment (723514372466) [reply tools/c7n_tencentcloud/c7n_tencentcloud/resources/mysql.py]: optimized

File: tools/c7n_tencentcloud/c7n_tencentcloud/resources/mysql_backup.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        resource_prefix = "instanceId"
+        taggable = True
+        datetime_fields_format = {
+            "Date": ("%Y-%m-%d %H:%M:%S", pytz.timezone("Asia/Shanghai"))
+        }
+
+    def augment(self, resources):
+        backup_resources = []
+        for resource in resources:
+            cli = self.get_client()
```
[the remark this answers is not in the document]
Comment (723514372466) [reply tools/c7n_tencentcloud/c7n_tencentcloud/resources/mysql_backup.py]: 😅，moved

File: tools/c7n_tencentcloud/c7n_tencentcloud/resources/mysql_backup.py (hunk 3 above)
Comment (723514372466) [reply tools/c7n_tencentcloud/c7n_tencentcloud/resources/mysql_backup.py:19]: After confirmation, the api is like this, if the instance has been deleted, the backup cannot be queried

Comment (thisisshi) [reply tools/c7n_tencentcloud/c7n_tencentcloud/resources/mysql_backup.py:19]: Are you able to open an upstream ticket to get that type of functionality? For now it's ok, given the constraints of the API but this doesn't really behave like any of the other cloud providers where backups can be queried without a parent resource

**Review by thisisshi — APPROVED** [review]

File: tools/c7n_tencentcloud/c7n_tencentcloud/resources/mysql_backup.py (hunk 3 above)
Comment (723514372466) [reply tools/c7n_tencentcloud/c7n_tencentcloud/resources/mysql_backup.py:19]: A requirement has been submitted. The product team follow it. 
I think we can merge this version and optimize it later when the new API is ready.

File: tools/c7n_tencentcloud/c7n_tencentcloud/resources/mysql_backup.py [hunk 7]
```diff
@@ -0,0 +1,40 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+from c7n_tencentcloud.provider import resources
+from c7n_tencentcloud.query import ResourceTypeInfo, QueryResourceManager
+from c7n_tencentcloud.utils import PageMethod, isoformat_datetime_str
+import pytz
+
+
+@resources.register("mysql-backup")
```
Comment (kapilt) [inline tools/c7n_tencentcloud/c7n_tencentcloud/resources/mysql_backup.py:9]: is this actually specific to mysql? afaics this api is the same across db types (mysql, Maria, postgresql, etc).

Comment (kapilt) [reply tools/c7n_tencentcloud/c7n_tencentcloud/resources/mysql_backup.py:9]: never mind it looks like each db type has a separate api endpoint.

**Pull request merged**